referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
+
+  var subresourceInvoker = {
+    "a-tag": queryLink,
+    "area-tag": queryAreaLink,
+    "fetch-request": queryFetch,
+    "iframe-tag": queryIframe,
+    "img-tag":  queryImage,
+    "script-tag": queryScript,
+    "worker-request": queryWorker,
+    "xhr-request": queryXhr
+  };
+
+  var pathForSubresource = SPEC_JSON["subresource_path"];
+
+  var referrerUrlResolver = {
+    "omitted": function() {
+      return undefined;
+    },
+    "origin": function() {
+      return location.protocol + "//" + location.hostname +
+             normalizePort(location.port) + "/";
+    },
+    "stripped-referrer": function() {
+      return stripUrlForUseAsReferrer(location.toString());
+    }
+  };
+
+  var t = {
+    _scenario: scenario,
+    _testDescription: testDescription,
+    _subresourceUrl: null,
+    _expectedReferrerUrl: null,
+    _constructSubresourceUrl: function() {
+      var domainForOrigin = {
+        "cross-origin": "www1." + location.hostname,
```

Comment:
http://wptserve.readthedocs.org/en/latest/pipes.html#sub gives you the ability to pull hosts via the `{{...}}` method (which you used below for ports). Why not use it here?